### PR TITLE
upgrade: Add error handling to the admin repo check page

### PR DESCRIPTION
**Pull request by Itxaka**

### CODE REVIEW

**Comment by Itxaka** [comment]
[(3) UI: Generic error handling (admin repo check)](https://trello.com/c/4eO0PsXW/206-3-ui-generic-error-handling-admin-repo-check)

File: assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js [hunk 1]
```diff
@@ -67,20 +68,24 @@
...
+                            }
                             return vm.repoChecks.checks[k].status === true
                         });
 
                         // if all the repos are ok, complete the step
                         if (vm.repoChecks.valid) {
                             upgradeStepsFactory.setCurrentStepCompleted();
+                        } else {
+                            // TODO(itxaka): This is a tmp thing, title and style should be clarified
+                            vm.error = { title: 'Some repositories are not ready', body: repoErrors.join(', ') };
```
Comment (skazi0) [inline assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js]: please move those texts to translations file (e.g. like here: https://github.com/crowbar/crowbar-ui/pull/115)

File: assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js [hunk 2]
```diff
@@ -67,20 +68,24 @@
...
                         }
                     },
                     // In case of failure
                     function (errorRepoChecksResponse) {
-                        // Expose the error list to repoChecks object
-                        // TODO(itxaka): Check the proper error response when the card is done
-                        // https://trello.com/c/chzg85j4/142-3-s49p7-error-reporting-on-crowbar-level
-                        vm.repoChecks.errors = errorRepoChecksResponse.data.errors;
+                        // TODO(itxaka): This is a tmp thing, title and style should be clarified
+                        vm.error = { title: 'Error while querying the API', body: errorRepoChecksResponse.data.errors }
```
Comment (skazi0) [inline assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js]: ditto + we need to decide on the format of `body` (string or what?)

**Review by skazi0 — CHANGES_REQUESTED** [review]

File: assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js (hunk 1 above)
Comment (Itxaka) [reply assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js]: Umm, I rather move them when we have the final text, that way we can have the TODO in there and its easily identifiable. We cannot set comments on the json file unfortunately, so Im worried that setting a temporal text in there may get missed. Thoughts?

Comment (skazi0) [reply assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js]: This or add the translations and create a followup card for updating the translations.

Comment (Itxaka) [reply assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js]: There is already a not-translated string merged as part of a previous PR, so I rather create a card to track this and change them when there is consensus on the titles/body style and so on.

Comment (skazi0) [reply assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js]: ok

**Review by skazi0 — APPROVED** [review]

File: assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js (hunk 1 above)
Comment (Itxaka) [reply assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js]: https://trello.com/c/Mpu84gVE/246-ui-error-messages-definitions

File: assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js (hunk 2 above)
Comment (Itxaka) [reply assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js]: same as above: https://trello.com/c/Mpu84gVE/246-ui-error-messages-definitions

**Comment by Itxaka** [comment]
Need to update to be in sync with https://github.com/crowbar/crowbar-ui/pull/115

**Comment by Itxaka** [comment]
@skazi0 updated to be in sync with your changes!

File: assets/app/features/upgrade/templates/administration-repositories-checks-page.jade [hunk 3]
```diff
@@ -16,3 +16,4 @@
             suse-lazy-spinner(delay="2000", active="upgradeAdminRepoChecksVm.repoChecks.running",
                 visible="upgradeAdminRepoChecksVm.repoChecks.spinnerVisible")
             span(translate='') upgrade.steps.admin-repository-checks.form.check-again
+suse-modal(error="upgradeAdminRepoChecksVm.error")
```
Comment (skazi0) [inline assets/app/features/upgrade/templates/administration-repositories-checks-page.jade]: please add `translation-prefix="upgrade.errors"` for proper error code scoping

File: assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js (hunk 1 above)
Comment (skazi0) [reply assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js]: After some discussion around the implementation of nodes repo checks, we decided that showing error popup for failed repos doesn't bring much benefit as the same information will be displayed in the main step page anyway (red marks).

File: assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js (hunk 2 above)
Comment (skazi0) [reply assets/app/features/upgrade/controllers/upgrade-administration-repositories-checks.controller.js]: After recent changes leading to unify error formats around the application, expected format for error modal input is the "common format" used in other places.
See: https://github.com/crowbar/crowbar-ui/pull/117 and https://github.com/crowbar/crowbar-ui/pull/119

**Review by skazi0 — CHANGES_REQUESTED** [review]
some adjustment is still needed to match recent changes in common error handling

**Comment by Itxaka** [comment]
@skazi0 updated! Thanks for the feedback!

File: assets/app/features/upgrade/templates/administration-repositories-checks-page.jade [hunk 4]
```diff
@@ -16,3 +16,4 @@
             suse-lazy-spinner(delay="2000", active="upgradeAdminRepoChecksVm.repoChecks.running",
                 visible="upgradeAdminRepoChecksVm.repoChecks.spinnerVisible")
             span(translate='') upgrade.steps.admin-repository-checks.form.check-again
+suse-modal(error="upgradeAdminRepoChecksVm.error", translation-prefix="upgrade.errors")
```
Comment (skazi0) [inline assets/app/features/upgrade/templates/administration-repositories-checks-page.jade]: s/error/errors/

**Review by skazi0 — CHANGES_REQUESTED** [review]

File: assets/app/features/upgrade/templates/administration-repositories-checks-page.jade (hunk 4 above)
Comment (Itxaka) [reply assets/app/features/upgrade/templates/administration-repositories-checks-page.jade]: wooops!

**Review by skazi0 — APPROVED** [review]

**Pull request merged**